msdsl/model.py

```diff
@@ -460,8 +464,15 @@ def set_from_sync_ram(self, signal: Union[AnalogSignal, str], format_: RealForma
         return self.add_assignment(SyncRamAssignment(signal=signal, format_=format_, addr=addr, clk=clk,
                                                      ce=ce, we=we, din=din, should_bind=should_bind))
 
-    def set_from_sync_func(self, signal: Union[Signal, str], func: GeneralFunction, in_: ModelExpr,
-                           clk=None, ce=None, rst=None, we=None, wdata=None, waddr=None):
+    def set_from_sync_func(self, *args, **kwargs):
+        return self.set_from_func(*args, func_mode='sync', **kwargs)
+
+    def set_from_async_func(self, *args, **kwargs):
```

Comment:
there are now asynchronous (clockless) and synchronous functions.  Synchronous functions are more resource efficient, but asynchronous may be more convenient in some cases.